test/two_links_moving_joint-launch.py

```diff
@@ -0,0 +1,56 @@
+import os
+import unittest
+
+from launch import LaunchDescription
+import launch
+from launch_ros.actions import Node
+import launch_testing
+from launch.substitutions import LaunchConfiguration
+from launch_ros.substitutions import FindPackageShare
+
+
+def generate_test_description(ready_fn):
+    test_exe_arg = launch.actions.DeclareLaunchArgument(
+        'test_exe',
+        description='Path to executable test',
+    )
+
+    process_under_test = launch.actions.ExecuteProcess(
+        cmd=[launch.substitutions.LaunchConfiguration('test_exe')],
+        output='screen',
+    )
+
+    urdf_file = os.path.join(os.path.dirname(__file__), 'two_links_moving_joint.urdf')
+    with open(urdf_file, 'r') as infp:
+        robot_desc = infp.read()
+
+    params = {'robot_description': robot_desc}
+    node_robot_state_publisher = Node(
+        package='robot_state_publisher',
+        node_executable='robot_state_publisher_node',
+        output='screen',
+        parameters=[params]
+    )
+
+    return LaunchDescription([
+        node_robot_state_publisher,
+        test_exe_arg,
+        process_under_test,
+        launch_testing.util.KeepAliveProc(),
+        launch.actions.OpaqueFunction(function=lambda context: ready_fn()),
```

Comment:
I think this is preferred now:
```suggestion
        launch_testing.actions.ReadyToTest(),
```

Reply:
So I tried this, but it makes the tests fail for reasons I don't understand.  I'll have to come back to it later to investigate.

Reply:
I figured it out.  There's another piece of magic I had to do to remove `ready_fn`.  Fixed in 9a6c78e47649fec084d66be1ac7caf3df39bd316.